psi4/driver/procrouting/diis.py

```diff
@@ -85,83 +122,97 @@ def copier(self, x, new_name):
         return copy
 
     def get_name(self, name, entry_num, item_num):
+        """ This is what we'll save an object to disk with."""
         return f"{self.name}: {name} Entry {entry_num}, Item {item_num}"
 
-    def get_dot_product(self, i, j):
+    def load_quantity(self, name, entry_num, item_num, force_new = True):
+        """ Load quantity from wherever it's stored, constructing a new object if needed. """
+        template_object = self.template[name][item_num]
+        if isinstance(template_object, float) or self.storage_policy == StoragePolicy.InCore:
+            quantity = self.stored_vectors[entry_num][name][item_num]
+            try:
+                quantity = quantity.clone()
+            except AttributeError:
+                # The quantity must have been a float. No need to clone.
+                pass
+        elif self.storage_policy == StoragePolicy.OnDisk:
+            entry_dims = template_object
+            full_name = self.get_name(name, entry_num, item_num)
+            psio = core.IO.shared_object()
+            if len(entry_dims) == 2:
+                quantity = core.Matrix(full_name, *entry_dims)
+                quantity.load(psio, psif.PSIF_LIBDIIS, core.SaveType.SubBlocks)
+            elif len(entry_dims) == 1:
+                quantity = core.Vector(full_name, *entry_dims)
+                quantity.load(psio, psif.PSIF_LIBDIIS)
+        else:
+            raise Exception(f"StoragePolicy {self.storage_policy} not recognized. This is a bug: contact developers.")
+
+        return quantity
+
+
+    def get_dot_product(self, i: int, j: int):
+        """ Get a DIIS dot product. i and j represent entry numbers. """
         key = frozenset([i, j])
         try:
             return self.cached_dot_products[key]
         except KeyError:
```

Comment:
```suggestion
        if key in self.cached_dot_products:
            return self.cached_dot_products[key]
        else:
```

Reply:
Rejected as needlessly verbose. Such usage of `try-except` is standard.